File: 128 Longest Consecutive Sequence/128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> hashmap;
        int max_len = 0;
        int curr_len = 0;

        for (int num : nums) {
            if (hashmap.find(num) == hashmap.end()) {
                int left = hashmap.find(num - 1) != hashmap.end() ? hashmap[num - 1] : 0;
                int right = hashmap.find(num + 1) != hashmap.end() ? hashmap[num + 1] : 0;

                curr_len = left + right + 1;
                hashmap[num] = curr_len;

                max_len = max(max_len, curr_len);

                if (left != 0) {
                    hashmap[num - left] = curr_len;
                }
                if (right != 0) {
                    hashmap[num + right] = curr_len;
                }
            }
        }
        return max_len;
    }
};

File: hashmap/128 Longest Consecutive Sequence/128.py
class Solution:
    def longestConsecutive(self, nums):
        table = {}
        max_len = 0

        for num in nums:
            if num not in table:
                left = table.get(num-1, 0)
                right = table.get(num+1, 0)

                curr_len = left + right + 1
                table[num] = curr_len

                max_len = max(curr_len, max_len)
                
                if left != 0: 
                    table[num - left] = curr_len
                    
                if right != 0:
                    table[num + right] = curr_len
                    
        return max_len